packages/radix-vue/src/Accordion/AccordionItem.vue

```diff
@@ -1,180 +1,119 @@
-  const orderedNodes = Array.from(
-    collectionNode.querySelectorAll("[data-radix-vue-collection-item]")
-  );
+const dataState = computed(() => {
+  console.log("datastate", open.value);
+  return open.value ? AccordionItemState.Open : AccordionItemState.Closed;
+});
 
-  return orderedNodes as HTMLElement[];
-}
+provide<AccordionItemProvideValue>(ACCORDION_ITEM_INJECTION_KEY, {
+  dataState,
+  disabled,
+  dataDisabled,
+  triggerId: useId(),
+});
 
-const handleKeyDown = (event: KeyboardEvent) => {
-  if (!ACCORDION_KEYS.includes(event.key)) return;
+const { primitiveElement, currentElement } = usePrimitiveElement();
 
-  const isDirectionLTR = accordionImplInjectedValue?.direction === "ltr";
-  const target = event.target as HTMLElement;
-  const triggerCollection = getItems().filter(
-    // (item) => !item.ref.current?.disabled
-    // (item) => !item.disabled
-    (item) => item
-  );
-  const triggerIndex = triggerCollection.findIndex(
-    // (item) => item.ref.current === target
-    (item) => item === target
+function handleArrowKey(e: KeyboardEvent) {
+  const newSelectedElement = useArrowNavigation(
+    e,
+    currentElement.value!,
+    injectedRoot?.parentElement.value!,
+    {
+      arrowKeyOptions: injectedRoot?.orientation,
+      dir: injectedRoot?.direction,
+    }
   );
-  const triggerCount = triggerCollection.length;
-
-  if (triggerIndex === -1) return;
-
-  // Prevents page scroll while user is navigating
-  event.preventDefault();
 
-  let nextIndex = triggerIndex;
-  const homeIndex = 0;
-  const endIndex = triggerCount - 1;
+  if (!newSelectedElement) return;
 
-  const moveNext = () => {
-    nextIndex = triggerIndex + 1;
-    if (nextIndex > endIndex) {
-      nextIndex = homeIndex;
-    }
-  };
-
-  const movePrev = () => {
-    nextIndex = triggerIndex - 1;
-    if (nextIndex < homeIndex) {
-      nextIndex = endIndex;
-    }
-  };
-
-  switch (event.key) {
-    case "Home":
-      nextIndex = homeIndex;
-      break;
-    case "End":
-      nextIndex = endIndex;
-      break;
-    case "ArrowRight":
-      if (accordionImplInjectedValue?.orientation === "horizontal") {
-        if (isDirectionLTR) {
-          moveNext();
-        } else {
-          movePrev();
-        }
-      }
-      break;
-    case "ArrowDown":
-      if (accordionImplInjectedValue?.orientation === "vertical") {
-        moveNext();
-      }
-      break;
-    case "ArrowLeft":
-      if (accordionImplInjectedValue?.orientation === "horizontal") {
-        if (isDirectionLTR) {
-          movePrev();
-        } else {
-          moveNext();
-        }
-      }
-      break;
-    case "ArrowUp":
-      if (accordionImplInjectedValue?.orientation === "vertical") {
-        movePrev();
-      }
-      break;
-  }
-
-  const clampedIndex = nextIndex % triggerCount;
-  triggerCollection[clampedIndex].focus();
-};
+  const trigger =
+    newSelectedElement.querySelector<HTMLButtonElement>("[aria-controls]");
```

Comment:
find another method to select the trigger. maybe put in inject as how other component does